webapp/app.py

```diff
@@ -69,21 +67,93 @@ def results():
     return flask.render_template("careers/results.html", **context)
 
 
-@app.route("/careers/admin", methods=["GET", "POST"])
-@app.route("/careers/all", methods=["GET", "POST"])
-@app.route("/careers/commercial-ops", methods=["GET", "POST"])
-@app.route("/careers/design", methods=["GET", "POST"])
-@app.route("/careers/engineering", methods=["GET", "POST"])
-@app.route("/careers/finance", methods=["GET", "POST"])
-@app.route("/careers/hr", methods=["GET", "POST"])
-@app.route("/careers/legal", methods=["GET", "POST"])
-@app.route("/careers/marketing", methods=["GET", "POST"])
-@app.route("/careers/project-management", methods=["GET", "POST"])
-@app.route("/careers/sales", methods=["GET", "POST"])
-@app.route("/careers/tech-ops", methods=["GET", "POST"])
-def department_group():
-    department = flask.request.path.split("/")[2]
-    vacancies = greenhouse_api.get_vacancies(department)
+# Class that collects department-specific content
+class Department(object):
+    __careers_directory = "./templates/careers"
+
+    def __parse_feed_department(feed_department):
+        field = {
+            "cloud engineering": "engineering",
+            "device engineering": "engineering",
+            "web and design": "design",
+            "web & design": "design",
+            "operations": "commercial-ops",
+            "human resources": "hr",
+            "techops": "tech-ops",
+            "product": "product management"
+        }
+
+        if feed_department.lower() in field:
+            return field[feed_department.lower()]
+
+        return feed_department
+
+    def __init__(self, name):
+        self.name = name
+        self.slug = Department.__parse_feed_department(name)
+        self.slug = self.slug.replace(" ", "-").lower()
+
+
+# Generates a list of departments
+def get_department_list():
+    departments = []
+    all_departments = greenhouse_api.get_departments()
+
+    """
+    Populate departments[] with Department objects,
+    ensuring that there are no duplicates
+    """
+    for item in all_departments:
+        new_department = Department(item)
+        is_new = True
+        for department in departments:
+            if department.slug == new_department.slug:
+                is_new = False
+                break
+        if is_new:
+            departments.append(new_department)
```

Comment:
I think this is good but could be improved by using a Set which de-dups automatically

Reply:
The departments list get indexed in some jinja templates, and to my knowledge sets don't support indexing?

Reply:
We could sort alphabetically each time right?